### PR TITLE
Save user progress

### PR DESCRIPTION
**Pull request by hdang32**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -14,14 +14,14 @@ const routes: Routes = [
 
   // { path: '', component: ScenarioListViewComponent },
   // { path: 'scenario/:id', component: ScenarioDetailComponent},
-  { path: ':roleId/scene/:sceneId', component: SceneContainerComponent },
-  { path: ':roleId/explanation/:sceneId', component: ExplanationViewComponent},
-  { path: 'role', component: RoleListViewComponent},
+  { path: ':userId/roles/:roleId/scenes/:sceneId', component: SceneContainerComponent },
```
Comment (jeannie3) [inline src/app/app-routing.module.ts]: Do we need to pass the userId in the route? I think we can just access it from localStorage right?

File: src/app/components/scene-container/scene-container.component.ts [hunk 2]
```diff
@@ -1,43 +1,72 @@
...
+  // unloadHandler(event) {
+  //   // ...
+  // }
+
+  @HostListener('window:beforeunload', ['$event'])
+  beforeUnloadHander(event) {
+    localStorage.setItem('closed', 'triggered')
+    alert("No no no");
+    event.preventDefault();
+  }
```
Comment (jeannie3) [inline src/app/components/scene-container/scene-container.component.ts]: What is this block of code supposed to do? Is it still needed or was this a test from before?

File: src/app/components/scene-container/scene-container.component.ts [hunk 3]
```diff
@@ -1,43 +1,72 @@
...
   allScenesForRole: Scene[]
   roleId: string;
+  userId: string;
+  progress: Progress;
+  isNew: boolean;
+
+  // @HostListener('window:unload', ['$event'])
+  // unloadHandler(event) {
+  //   // ...
+  // }
```
Comment (jeannie3) [inline src/app/components/scene-container/scene-container.component.ts]: Comment

File: src/app/components/scene-container/scene-container.component.ts [hunk 4]
```diff
@@ -1,43 +1,72 @@
...
-      this.router.navigate([this.roleId + '/scene/' + nextScene]);
-    }
+    this.scenarioService.getProgress(this.userId, this.roleId).subscribe(progress => {
+      if (progress.length === 0) {
+        this.progress = new Progress();
+        this.progress.user_id = this.userId;
+        this.progress.role_id = this.roleId;
+        this.progress.scene_id = this.currentScene.id;
+        this.isNew = true;
+        console.log("new progress " + JSON.stringify(this.progress));
```
Comment (jeannie3) [inline src/app/components/scene-container/scene-container.component.ts]: console.log

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (hdang32) [reply src/app/app-routing.module.ts]: Yeah just wanna change the route so we know which user is accessing.

File: src/app/components/scene-container/scene-container.component.ts (hunk 2 above)
Comment (hdang32) [reply src/app/components/scene-container/scene-container.component.ts]: should be removed

**Pull request merged**